app/instance-initializers/ember-i18n.js

```diff
@@ -14,6 +14,7 @@ export default {
     const key = 'service:i18n';
     const i18n = instance.lookup ? instance.lookup(key) : instance.container.lookup(key);
     i18n.set('locale', defaultLocale);
+    i18n.set('allowLocaleOverride', (ENV.i18n || {}).allowLocaleOverride);
```

Comment:
Because this initializer doesn't run in all situations (notably non-integration tests), I suggest moving this logic into the Service itself:
```js
allowLocaleOverride: Ember.computed({
  get() { return (ENV.i18n || {}).allowLocaleOverride; },
  set(prop, newValue) { return newValue; }
})
```